ledger/participant-integration-api/src/main/scala/platform/apiserver/StandaloneApiServer.scala

```diff
@@ -165,8 +167,15 @@ final class StandaloneApiServer(
       .fold({ case (err, file) => sys.error(s"Could not load package $file: $err") }, identity)
   }
 
-  private def writePortFile(port: Port): Unit =
-    config.portFile.foreach { path =>
-      PortFiles.write(path, port)
+  private def writePortFile(port: Port): Future[Unit] = {
```

Comment:
Does the change to return a future matter or is the only crucial change to throw an exception? No objection to keep it just trying to understand the bug.

Reply:
The only crucial part is throwing the exception. The future is just there to make this fit into the resource owner for-block on the calling side. It seemed cleaner to me to integrate it into that block rather than to have a line in the yield part throw an exception.

Reply:
You could use a `Try` and then use `ResourceOwner.forTry` on the other side.